docs/mkdocs/secureboot-signing.md

```diff
@@ -0,0 +1,75 @@
+# Signing kernel modules using KMM
+
+### Secureboot with KMM
+
+For more details on using Secureboot see [here](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/managing_monitoring_and_updating_the_kernel/signing-kernel-modules-for-secure-boot_managing-monitoring-and-updating-the-kernel) or [here](https://wiki.debian.org/SecureBoot)
+
+### Using Signing with KMM
+
+On a secureboot enabled system all kernel modules (kmods) must be signed with a public/private keypair enrolled into the Machine Owner's Key or MOK database. Drivers distributed as part of a distribution should already be signed by the distributions private key, but for kernel modules build out-of-tree KMM supports signing kernel modules using the "sign" section of the kernel mapping.
+
+To use this functionality you need:
+
+ * A public private key pair in the correct (der) format
+ * At least one secureboot enabled nodes with the public key enrolled in its MOK database
+ * A driver container image 
+
+The yaml below will add the public/private keypair as secrets with the required key names ("key" for the private key, "cert" for the public key). It will then pull down the `unsignedImage` image, open it up, sign the kernel modules listed in `filesToSign`, add them back anbd push the resulting image as `containerImage`. 
+
+KMM should then deploy the daemonset that loads the signed kmods onto all the nodes with thati match the selector.
+The driver containers should run successfully on any nodes that have the public key in their MOK databse, and any nodes that are not secureboot enabled (which will just ignore the signature). They should fail to load on any that have secureboot enabled but do not have that key in their mok database.
+
+If your driver containers end up in `PostStartHookError` or `CrashLoopBackOff` status, and `kubectl describe` shows an event: `modprobe: ERROR: could not insert 'simple_kmod': Required key not available` then the kmods are eitehr not signed, or signed with the wrong key.
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-signing-key-pub
+  namespace: default
+type: Opaque
+data:
+  cert: <base64 encoded secureboot public key>
+
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-signing-key
+  namespace: default
+type: Opaque
+data:
+  key: <base64 encoded secureboot private key>
+
+---
+apiVersion: kmm.sigs.k8s.io/v1beta1
+kind: Module
+metadata:
+  name: kmmo-minimal-kmod-combine
+spec:
+  moduleLoader:
+    serviceAccountName: default
+    container:
+      modprobe:
+        # the name of the kmod to load
+        moduleName: 'simple-kmod'
+      kernelMappings:
+        # the kmods will be deployed on all nodes in the cluster with a kernel that matches the regexp
+        - regexp: '^.*\.x86_64$'
+          # the container to produce containing the signed kmods
+          containerImage: <quay.io/myuser/my-driver:signed>
```

Comment:
I guess the tag should always be the kernel version? did you mean `my-driver-signed:<kernel version>`?

Reply:
I don't think we enforce the tag having to be kernel-version, but it is definitely a good idea and docs should  promote best practice so I've updated it.